src/ReportManager/DetailsSection/SchemaForm/utils/useSchemaValidations/index.js

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useMemo } from 'react';
+import Ajv2020 from 'ajv/dist/2020';
+import { useTranslation } from 'react-i18next';
+
+const ajv = new Ajv2020({ allErrors: true });
+
+const useSchemaValidations = (schema) => {
+  const { t } = useTranslation('reports', { keyPrefix: 'reportManager.detailsSection.schemaForm.errors' });
+
+  const validate = useMemo(() => ajv.compile(schema.json), [schema.json]);
+
+  const runValidations = useCallback((formData) => {
+    const valid = validate(formData);
+
+    if (!valid) {
+      const fieldErrors = validate.errors.reduce((accumulator, error) => {
+        if (error.keyword === 'required') {
+          return { ...accumulator, [error.params.missingProperty]: t('required') };
```

Comment:
So, by using `error.params.missingProperty` for the error mapping it means We are going to have the same generic error message for different field types, right? 
I'm just asking, because idk if there might be some fields requiring more custom error messages ?

Reply:
It means we will have the same error message for the same type of error. In this case, it is the error required. Of course, we can add more complexity to the condition that renders the text, in case there is a very specific scenario for some kind of field that wants a different message for the same error. We can totally handle that by passing the field type to the hook 😄  But honestly, I don't think that will happen much, or maybe even at all.